R/generate.R

```diff
@@ -250,22 +251,34 @@ permute <- function(x, reps = 1, variables, ..., call = caller_env()) {
 
 permute_once <- function(x, variables, ..., call = caller_env()) {
   dots <- list(...)
+  null <- attr(x, "null")
+
+  if (!is_hypothesized(x) ||
+      !null %in% c("independence", "paired independence")) {
+     abort(
+        paste0("Permuting should be done only when doing independence ",
+               "hypothesis test. See `hypothesize()`."),
+        call = call
+     )
+  }
 
-  if (is_hypothesized(x) && (attr(x, "null") == "independence")) {
-    # for each column, determine whether it should be permuted
-    needs_permuting <- colnames(x) %in% process_variables(variables, FALSE)
-
-    # pass each to permute_column with its associated logical
-    out <- purrr::map2(x, needs_permuting, permute_column)
-    out <- tibble::new_tibble(out)
+  variables <- process_variables(variables, FALSE)
+  if (null == "independence") {
+     # for each column, determine whether it should be permuted
+     needs_permuting <- colnames(x) %in% variables
 
-    copy_attrs(out, x)
+     # pass each to permute_column with its associated logical
+     out <- purrr::map2(x, needs_permuting, permute_column)
+     out <- tibble::new_tibble(out)
   } else {
-     abort(paste0(
-      "Permuting should be done only when doing independence hypothesis test. ",
-      "See `hypothesize()`."
-    ), call = call)
+     out <- x
+     signs <- sample(c(-1, 1), nrow(x), replace = TRUE, prob = c(.5, .5))
```

Comment:
Should `signs` be `sample()`d just once and then its values permuted with every replicate, or sampled with each replicate? Or is the un-permuted `signs` just `sign(diff)`?

Reply:
Okay, perfect!